syntax = "proto2";
package deepsmith;

message ServiceStatus {
  enum ReturnCode {
    SUCCESS = 0;
    NOT_IMPLEMENTED_ERROR = 1;
    INTERNAL_ERROR = 2;
    INVALID_REQUEST_PARAMETERS = 3;
  }

  // The name of the server sending the response.
  optional string client = 1 [default = "<anon>"];

  // The return code of the response.
  optional ReturnCode returncode = 2 [default = NOT_IMPLEMENTED_ERROR];

  // If status != SUCCESS, a message describing the error.
  optional string error_message = 3;

  // If status != SUCCESS, whether the client should send a new request for the
  // same operation, or abort.
  optional bool should_retry = 4;
}

message ServiceConfig {
  optional string hostname = 1;
  optional int32 port = 2;
}
